docs/DeveloperGuide.md

```diff
@@ -1,34 +1,270 @@
+
+<p align="center">
+  <img alt="PERT Chart" src="diagrams/out/PERT_Chart.png" />
+  <br><em>Figure #</em>
+</p>
+
+#### <a id="notus"><ins>2.2 NotUS</ins></a>
+
+NotUs manages the flow of the application. On launch, it will create the necessary components, as listed above and then attempts to load any existing saved files into the application. Subsequently, it will accept and interpret the user input and execute the commands accordingly. The diagram below depicts the main flow of the application.
+
+<p align="center">
+  <img alt="NotUS" src="diagrams/out/Notus.png" />
+  <br><em>Figure #</em>
+</p>
+
+#### <a id="parserManager"><ins>2.3 ParserManager</ins></a>
+
+The ParserManager manages the creation of specific parser objects based on the type of command. The parser then makes sense of the user input and calls the respective commands into action.
+
+1. Receives the user input message as a whole.
+1. Interprets the type of command and creates the respective parser for each command.
+1. The parser then splits the message to identify all the parameters provided.
+1. Creates and returns the Command class respectively.
+ 
+<p align="center">
+  <img alt="Parser" src="diagrams/out/Parser.png" />
+  <br><em>Figure #</em>
+</p>
+
+💡 Note that the alternate paths in the sequence diagram above are not exhaustive. There is an alternate path for each unique command. As there are too many paths, they are omitted from the diagram. The Command objects in the diagram are used to represent a generic Command object that is created through the Parser. Refer to the next figure for more details.
+ 
+ <p align="center">
+   <img alt="Parser" src="diagrams/out/AddNoteParser.png" />
+   <br><em>Figure #</em>
+ </p>
+
+Based on the user input, the Parser handles and creates the corresponding Command object.
+
+#### <a id="commands"><ins>2.4 Commands</ins></a>
+
+**AddNoteCommand**
+
+1. Created by the parser function
+1. Gets the note with all its variables prepared in ParseAddNoteCommand. 
+1. Obtain content input into note.
+1. Process and stores tags into TagsManager.
+1. Handle saving of notes.
+1. Returns the title, tags as well as the contents of the note. 
+
+<p align="center">
+   <img alt="Parser" src="diagrams/out/AddNote_Sequence.png"/>
+   <br><em>Figure #</em>
+</p>
+
+**PinCommand**
+
+1. Created by the parser function
+1. Gets the note that is referenced too either by title or index 
+1. Toggles the pinned status of the specified note. 
+1. Returns the title as well as the pinned status of the note. 
+
+<p align="center"> 
+   <img alt="PinCommand" src="diagrams/out/PinCommand.png"/>
```

Comment:
Is there supposed to be anything inside the square brackets in this alt block?
![image](https://user-images.githubusercontent.com/60461477/97397557-7f49eb80-1924-11eb-8c89-e34b6a16640b.png)

Reply:
Is `PinByIndex` a flag, because in case of opt the square brackets should describe a condition.